File: Back/.Net/BetFriends.Infrastructure/Outbox/Outbox.cs
using BetFriends.Application.Abstractions;

namespace BetFriends.Infrastructure.Outbox
{
    public class Outbox
    {
        public Outbox(string type, string data, DateTime occurredOn)
        {
            Id = Guid.NewGuid();
            OccurredOn = occurredOn;
            Type = type;
            Data = data;
        }

        public Guid Id { get; }
        public DateTime OccurredOn { get; }
        public DateTime? ProcessedOn { get; private set; }
        public string Type { get; }
        public string Data { get; }

        public void Handled(IDateProvider dateProvider)
        {
            ProcessedOn = dateProvider.GetDate();
        }
    }
}

File: Back/NodeJS/modules/shared/infrastructure/outbox/Outbox.ts
import { IDateTimeProvider } from "../../domain/IDateTimeProvider";

export class Outbox {
    HandledAt?: Date;
    Handled(dateProvider: IDateTimeProvider) {
        this.HandledAt = dateProvider.GetDate()
    }
    constructor(public Id: string, public Type: string, public Data: string, public CreatedAt: Date) {
        
    }
}